lambda/save-raw-events/save-raw-events.ts

```diff
@@ -0,0 +1,92 @@
+import { SQSEvent } from "aws-lambda";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  PutCommandOutput,
+} from "@aws-sdk/lib-dynamodb";
+import {
+  SendMessageCommand,
+  SendMessageRequest,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import { TxmaEvent, UserData } from "./models";
+
+const { TABLE_NAME } = process.env;
+const marshallOptions = {
+  convertClassInstanceToMap: true,
+};
+const translateConfig = { marshallOptions };
+
+const dynamoClient = new DynamoDBClient({});
+const dynamoDocClient = DynamoDBDocumentClient.from(
+  dynamoClient,
+  translateConfig
+);
+const sqsClient = new SQSClient({});
+const { DLQ_URL } = process.env;
+
+const getEventId = (): string => {
+  return `${Date.now()}-${Math.floor(Math.random() * 100000)}`;
+};
+
+const getTTLDate = (): number => {
+  const SECONDS_IN_AN_DAY = 60 * 60 * 24;
+  const secondsSinceEpoch = Math.round(Date.now() / 1000);
+  const expirationTime = secondsSinceEpoch + 90 * SECONDS_IN_AN_DAY;
+  return expirationTime;
+};
+
+export const validateUser = (user: UserData): void => {
+  if (user.user_id === undefined) {
+    throw new Error(`Could not find User ${user}`);
+  }
+};
+
+export const validateTxmaEventBody = (txmaEvent: TxmaEvent): void => {
+  if (
+    txmaEvent.client_id !== undefined &&
+    txmaEvent.timestamp !== undefined &&
+    txmaEvent.event_name !== undefined &&
+    txmaEvent.component_id !== undefined &&
+    txmaEvent.user !== undefined
+  ) {
+    validateUser(txmaEvent.user);
+  } else {
+    throw new Error(`Could not validate UserServices ${txmaEvent}`);
+  }
```

Comment:
I wonder if the existence of this here means we could lose some of this validation from the query step?
Or if once this is merged that's an opportunity to explore DRY-ing this out?

Reply:
Oh yes makes sense . I can do that after this is tested and all works fine, if that's alright with you .